### PR TITLE
IRIS-4153 | Bypass mw.message and use format.js convention in OSN

### PR DESCRIPTION
**Pull request by rogatty**
## Links
https://wikia-inc.atlassian.net/browse/IRIS-4153

## Description
Follow up to https://wikia-inc.atlassian.net/browse/IRIS-4937, will be followed up with https://wikia-inc.atlassian.net/browse/IRIS-4975.

## Reviewers

@slayful 
/cc @hakubo @TK-999

### CODE REVIEW

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.text-formatter.js [hunk 1]
```diff
@@ -15,10 +15,15 @@ define('ext.wikia.design-system.on-site-notifications.text-formatter', [
 
 			function fillArgs(message, args) {
 				return Object.keys(args).reduce(function (acc, key) {
-					return acc.replace('__' + key + '__', args[key])
+					return acc.replace('{' + key + '}', args[key])
 				}, message);
 			}
 
+			function getMessage(key) {
+				// TODO IRIS-4975 use mw.message(key).parse()
```
Comment (WikiaTech) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.text-formatter.js:23]: INFO Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.text-formatter.js [hunk 2]
```diff
@@ -15,10 +15,15 @@ define('ext.wikia.design-system.on-site-notifications.text-formatter', [
...
+				return window.mw.message().map.values[key];
```
Comment (mszabo-wikia) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.text-formatter.js]: `mw.messages.get(key)`?

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Review by slayful — APPROVED** [review]
LGTM

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**